python/test/test_api.py

```diff
@@ -0,0 +1,238 @@
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import asyncio
+import queue
+import time
+import unittest
+
+import numpy
+import pytest
+import tritonserver
+
+try:
+    import cupy
+except Exception:
+    cupy = None
+
+
+class TrtionServerAPITest(unittest.TestCase):
+    def test_not_started(self):
+        server = tritonserver.Server()
+        with self.assertRaises(tritonserver.InvalidArgumentError):
+            server.is_ready()
+
+    @pytest.mark.skipif(cupy is None, reason="Skipping gpu memory, cpupy not installed")
+    def test_gpu_memory(self):
+        import cupy
+
+        server = tritonserver.Server(
+            model_repository="/workspace/models", exit_timeout=5
+        )
+
+        server.start(blocking=True)
+
+        test = server.get_model("test")
+        fp16_input = cupy.array([[5], [6], [7], [8]], dtype=numpy.float16)
+        responses = test.infer(inputs={"fp16_input": fp16_input}, request_id="1")
+
+        for response in responses:
+            print(response)
+
+        responses = server.models["test"].infer(
+            inputs={"fp16_input": fp16_input}, request_id="1"
+        )
+
+        for response in responses:
+            print(response)
+        try:
+            pass
+        #            server.stop()
+        except Exception as error:
+            print(error)
+
+    def test_unload(self):
+        server = tritonserver.Server(
+            model_repository="/workspace/models",
+            exit_timeout=5,
+            model_control_mode=tritonserver.ModelControlMode.EXPLICIT,
+            startup_models=["test"],
+            log_verbose=True,
+            log_error=True,
+            log_info=True,
+        )
+        server.start(blocking=True)
+
+        model = server.models["test"]
+
+        responses = model.infer(
+            inputs={"fp16_input": numpy.array([[0.5]], dtype=numpy.float16)}
+        )
+
+        print(list(responses)[0])
+
+        print(model.is_ready())
+
+        model_versions = [key for key in server.models.keys() if key[0] == model.name]
+
+        server.unload_model(model.name, blocking=True)
+
+        server.unload_model("foo", blocking=True)
+
+        #        del responses
+
+        while True:
+            if [
+                key
+                for key in model_versions
+                if (
+                    server.models[key].state not in server.models[key].state is not None
+                    and server.models[key].state != "UNAVAILABLE"
+                )
+            ]:
+                print(list(server.models.items()))
+                time.sleep(5)
+                continue
+            break
+        print(server.models[model.name])
+        print(list(server.models.items()))
+
+        print(model.is_ready())
+        server.stop()
```

Comment:
Will this be called implicitly when object goes out of scope?

Reply:
yes